tkweb/apps/gallery/models.py

```diff
@@ -160,6 +161,34 @@ def clean(self):
             else:
                 self.slug = self.date.strftime('%Y%m%d%H%M%S_%f')[:len("YYYYmmddHHMMSS_ff")]
 
+    def save(self, *args, **kwargs):
+        """
+        This saves the Image, tries to prewarm VersatileImageField and deletes
+        itself again if it fails. Ideally this would be done in clean(), but
+        VersatileImageField cannot prewarm before it is saved and does not have
+        a clean that checks if the warming is bound to succeed.
+
+        """
+        super().save(*args, **kwargs)
+        image_warmer = VersatileImageFieldWarmer(
+            instance_or_queryset=self, rendition_key_set="gallery", image_attr="file"
+        )
+
+        num_created, failed_to_create = image_warmer.warm()
+        if failed_to_create:
+
+            self.delete()  # Hey! Look at me!
+
+            logger.warning(
+                "Prewarming during save() of %s failed. Deleting again." % self
+            )
+            raise ValidationError("Corrupt image. Deleting")
+
+        logger.debug(
+            "generateImageThumbnails: %d thumbnails created. Missing %s"
+            % (num_created, failed_to_create)
+        )
```

Comment:
Jeg synes det er en dårlig idé at have logging på en ikke-fejlende udførsel af Image.save(). Desuden er `failed_to_create` altid 0 her. Antallet af thumbnails der oprettes for et enkelt billede vil vel altid være det samme (undtaget når vi ændrer vores thumbnail settings), så jeg tror heller ikke at log-beskeden er så nyttig?

Reply:
Det kommer fra den gamle kode i `generateImageThumbnails()` længere nede. (Som burde have være slettet i dette commit.)  Jeg er enig. Den var brugbar da det var den eneste log besked fra `warm()`, men alt brugbar info er i beskeden over.